includes/functions/functions_dates.php

```diff
@@ -195,34 +202,40 @@ function zen_checkdate($date_to_check, $format_string, &$date_array)
 
         $size = count($format_string_array);
         for ($i = 0; $i < $size; $i++) {
-            if ($format_string_array[$i] == 'mm' || $format_string_array[$i] == 'mmm') $month = $date_to_check_array[$i];
-            if ($format_string_array[$i] == 'dd') $day = $date_to_check_array[$i];
-            if (($format_string_array[$i] == 'yyyy') || ($format_string_array[$i] == 'aaaa')) $year = $date_to_check_array[$i];
+            if ($format_string_array[$i] === 'mm' || $format_string_array[$i] === 'mmm') {
+                $month = $date_to_check_array[$i];
+            }
+            if ($format_string_array[$i] === 'dd') {
+                $day = $date_to_check_array[$i];
+            }
+            if (($format_string_array[$i] === 'yyyy') || ($format_string_array[$i] === 'aaaa')) {
+                $year = $date_to_check_array[$i];
+            }
         }
     } else {
-        if (strlen($format_string) == 8 || strlen($format_string) == 9) {
-            $pos_month = strpos($format_string, 'mmm');
-            if ($pos_month != false) {
-                $month = substr($date_to_check, $pos_month, 3);
-                $size = count($month_abbr);
-                for ($i = 0; $i < $size; $i++) {
-                    if ($month == $month_abbr[$i]) {
-                        $month = $i;
-                        break;
-                    }
+        if (!in_array(strlen($format_string), [8, 9])) {
+            return false;
+        }
+
+        $pos_month = strpos($format_string, 'mmm');
+        if ($pos_month != false) { // Should this be not exactly equal? Found at 0 is ok?
```

Comment:
Yes, found-at-position-zero is a legitimate possibility.

Reply:
I'll remove that comment; it was carried over from the original PR.